File: shuffle-the-array.py
from typing import List

class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
      list = []
      halfwayPoint = len(nums) // 2
      for i in range(halfwayPoint):
        list.append(nums[i])
        list.append(nums[i + halfwayPoint])
      return list

File: shuffle-the-array.rs
impl Solution {
    pub fn shuffle(nums: Vec<i32>, n: i32) -> Vec<i32> {
        let frontSlice = &nums[0..n as usize];
        let backSlice = &nums[n as usize..nums.len() as usize];

        let mut v = vec![];

        for l in 0..n {
            v.push(frontSlice[l as usize]);
            v.push(backSlice[l as usize]);
        }

        return v;
    }
}
